### DISCUSSION TITLE
FR - Allow Admin Tables to be sorted alphabetically

### DISCUSSION THREAD

**Discussion by engram-design** [craftcms/cms#8773] [Ideas]
It'd be great if Admin Tables in the CP could have their column headings clicked on to toggle sorting items alphabetically. Of course, we'd need to consider this behaviour on admin table's that can have a specific, manual order.

In all instances, this would be visual, as opposed to a saved order. Should be a little more manageable with Vue at least.

Refer https://github.com/verbb/navigation/issues/130#issuecomment-580175430

**Comment by nfourtythree** [comment]
Hey @engram-design 

This is actually possible already if you provide your data via the `tableDataEndpoint` param. Using this gives you the ability to sort the data in a controller action. An example of how this is done can be seen in the [Commerce Customers table action endpoint](https://github.com/craftcms/commerce/blob/develop/src/controllers/CustomersController.php#L104-L241) and the [email column in the template](https://github.com/craftcms/commerce/blob/develop/src/templates/customers/_index.html#L20).

We will look into supporting sorting when simply providing data directly. However, this poses some challenges as we do not know what data is being provided in each column.

Thanks!
